### PR TITLE
IPI private cluster on alibabacloud - insert the VPC settings in install-config

### PR DESCRIPTION
**Pull request by jianli-wei**
Cont. to https://github.com/openshift/release/pull/31664.

1. The VPC settings, i.e. `platform.alibabacloud.vpcID` and `platform.alibabacloud.vswitchIDs`, should be inserted into install-config.yaml, so added step `ipi-conf-alibabacloud-provisioned-network` accordingly. 
2. Update step `gather-alibabacloud-console` to enable client proxy when applicable.
3. Update step `alibabacloud-provision-vpc` to sleep a bit after `AssociateEipAddress`, and add stage logs.
4. Update step `alibabacloud-provision-bastionhost` to sleep a bit more after `RunInstances`, and add stage logs.

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

File: ci-operator/step-registry/ipi/conf/alibabacloud/ipi-conf-alibabacloud-commands.sh [hunk 1]
```diff
@@ -33,3 +33,7 @@ platform:
     region: ${REGION}
     resourceGroupID: ${CI_RESOURCE_GROUP_ID}
 EOF
+
+if [[ -s "${SHARED_DIR}/customer_vpc_subnets.yaml" ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/alibabacloud/ipi-conf-alibabacloud-commands.sh:37]: Suggest to introduce a dedicated step to do the customer vpc setting, similar to https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/private/provision/cucushift-installer-rehearse-azure-ipi-private-provision-chain.yaml#L9

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/alibabacloud/ipi-conf-alibabacloud-commands.sh:37]: I meant we can introduce `ipi-conf-alibabacloud-provisionednetwork` step to test the customer vpc case.

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

File: ci-operator/step-registry/ipi/conf/alibabacloud/ipi-conf-alibabacloud-ref.yaml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 ref:
   as: ipi-conf-alibabacloud
-  from: tools
+  from_image:
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/alibabacloud/ipi-conf-alibabacloud-ref.yaml:3]: I guess we do not need this change any more.

File: ci-operator/step-registry/ipi/conf/alibabacloud/provisioned-network/OWNERS [hunk 3]
```diff
@@ -0,0 +1,4 @@
+approvers:
+- alibaba-approvers
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/alibabacloud/provisioned-network/OWNERS:2]: I guess we can add our QE member here too.

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-alibabacloud

**Pull request merged**